jormungandr/src/fragment/pool.rs

```diff
@@ -67,40 +71,54 @@ impl Pool {
             .zip(fragments_exist_in_logs)
             .filter(|(_, exists_in_logs)| !exists_in_logs)
             .map(|(fragment, _)| fragment);
-        let new_fragments = self.pool.insert_all(new_fragments);
-        let count = new_fragments.len();
-        debug!(
-            self.logger,
-            "{} of the received fragments were added to the pool", count
-        );
-        let fragment_logs = new_fragments
-            .iter()
-            .map(move |fragment| FragmentLog::new(fragment.id(), origin))
-            .collect::<Vec<_>>();
+
+        let mut max_added = 0;
+
+        for (i, pool) in self.pools.iter_mut().enumerate() {
+            let new_fragments = pool.insert_all(new_fragments.clone());
+            let count = new_fragments.len();
+            debug!(
+                self.logger,
+                "{} of the received fragments were added to the pool number {}", count, i
+            );
+            let fragment_logs = new_fragments
+                .iter()
+                .map(move |fragment| FragmentLog::new(fragment.id(), origin))
+                .collect::<Vec<_>>();
+            self.logs.insert_all(fragment_logs);
+            if count > max_added {
+                max_added = count;
+            }
+        }
+
         for fragment in new_fragments.into_iter() {
             let fragment_msg = NetworkMsg::Propagate(PropagateMsg::Fragment(fragment));
             network_msg_box
                 .send(fragment_msg)
                 .await
                 .map_err(Error::CannotPropagate)?;
         }
-        self.logs.insert_all(fragment_logs);
-        Ok(count)
+
+        Ok(max_added)
     }
 
     pub fn remove_added_to_block(&mut self, fragment_ids: Vec<FragmentId>, status: FragmentStatus) {
-        self.pool.remove_all(fragment_ids.iter().cloned());
+        for pool in &mut self.pools {
+            pool.remove_all(fragment_ids.iter().cloned());
```

Comment:
Hmm, why would we need to take ownership over the IDs in that iterator argument...

Reply:
No point in that actually.